tidy3d/plugins/resonance/resonance.py

```diff
@@ -0,0 +1,272 @@
+"""Find resonances in time series data
+"""
+
+from typing import Tuple
+from functools import partial
+
+import numpy as np
+import scipy.linalg
+import xarray as xr
+
+from pydantic import Field
+
+from ...components.base import Tidy3dBaseModel
+from ...components.data.data_array import ScalarFieldTimeDataArray
+from ...constants import HERTZ
+from ...log import SetupError
+
+# ResonanceData will be used internally
+class ResonanceData(Tidy3dBaseModel):
+    """Data class for storing objects computed while running the solver."""
+
+    eigvals: np.ndarray[complex] = Field(
+        ..., title="Eigenvalues", description="Resonance eigenvalues."
+    )
+    complex_amplitudes: np.ndarray[complex] = Field(
+        None, title="Complex amplitudes", description="Complex resonance amplitudes"
+    )
+    errors: np.ndarray[float] = Field(
```

Comment:
type annotations like this seem to fail in python 3.7.  Let's use `ArrayLike[complex]` or if that doesn't work, just `np.ndarray` without specifying the dtype is ok for now.